odxtools/diaglayer.py

```diff
@@ -238,12 +273,59 @@ def ecu_variant_patterns(self) -> List[EcuVariantPattern]:
     #######
     # <stuff subject to value inheritance>
     #######
+    @property
+    def diag_comms(self) -> NamedItemList[Union[DiagService, SingleEcuJob]]:
+        """All diagnostic communication primitives applicable to this DiagLayer
+
+        Diagnostic communication primitives are diagnostic services as
+        well as single-ECU jobs. This list has all references
+        resolved.
+        """
+        return self._diag_comms
```

Comment:
Missing setting initial values of those private fields in __init__

Reply:
they are created in `._finalize_init()`. I'm not really happy about the visibility of this solution, but given that this is a private implementation detail, `mypy` does not complain about it, and that this is a pretty common pattern in odxtools, I don't want to fix that in this PR. (as I see it, the main issue with the current solution is that accessing `diag_layer.diag_comms` will raise `AttributeError` if the `diag_layer` has not been properly initialized which is not very intuitive. That said, I guess that "normal" users will never encounter this?)